### DISCUSSION TITLE
LFRic Apps vn2.0 Release

### DISCUSSION THREAD

**Discussion by james-bruten-mo** [MetOffice/simulation-systems#273] [Announcements]
Dear All,

Thank you for your efforts reviewing and developing LFRic Apps vn1.2 - a release of this is imminent.

In addition to vn1.2 we have a number of technical changes to the code base that are near completion and we would like to have released on timescales shorter than the next expected release around March time. As such we are proposing to complete an interim release of LFRic Apps by mid-December. The technical changes that we hope to include in this are detailed below - as some of these are non-backwards compatible we plan on making this a major version change to vn2.0. During this reasonably short release window we will accept other changes for commit, however we will release when the desired technical changes are on. We will not be holding this release for any other tickets. The main disruption we see to LFRic developers is the need to rebranch from vn1.2 to vn2.0 when the release is done.

The changes expected are:

**Psyclone 3 Upgrade**:

PSyclone 3.0.0 release will deliver new functionality for supporting LFRic science and performance optimisation (detailed information will be published here https://github.com/stfc/PSyclone/releases). There are, however, some non-backward compatible changes in PSyclone LFRic API which impacts how LFRic Apps and Core are built and how they configure PSyclone.  More detailed guidance on impacts and how to navigate the transition will be provided nearer to the upgrade.

**Introduction of Upgrade Macros**:

The process for working with Upgrade Macros in LFRic has been developed over the last few months. These will allow upgrading of rose configurations in both the test suite and scientific suites more easily than is currently the case. The introduction of Upgrade Macros will involve some changes to working practices - further details about these changes will follow, including advice for both developers and reviewers. We also plan to run a session demonstrating their use in an LFRic context (for those familiar with UM or Jules macros, the process is slightly more complex due to the distributed nature of LFRic metadata).

**Porting to EX machines**:

Work is ongoing to port the LFRic models to the new supercomputers in time for UAT expected early in the new year. This work involves adding the new platform to the test suite and code changes to address issues found when running with the newer compilers on the new machines.

If you have any questions or concerns, please get in touch with us, or the Simulation IT teams.

James, Iva and Jenny on behalf of Sim IT

**Comment by jennyhickson** [comment]
Work is well underway for vn2.0 and we're currently on target to release this mid-December. 

Please do keep vn2.1 and "next_release" (which can be committed at any time) tickets flowing, and please remember to use branches created from vn1.2 for all upcoming developments. 

Thanks!

**Reply by dennissergeev** [reply]
Thanks for the update. Could you clarify what you mean by the last point? Should I not upgrade my branch to vn2.0 when it's released?

**Reply by jennyhickson** [reply]
Sorry for the confusion. Upgrading to vn2.0 will be needed once that release is out. My point was to encourage people to use branches from a revision as we've had a few tickets for review recently that haven't done so and its caused problems. 

Given the tickets currently in review, UM vn13.7, JULES vn7.7 etc will continue to work with LFRic Apps vn2.0, so linked changes will only need to upgrade their LFRic Apps branch.

**Reply by dennissergeev** [reply]
OK many thanks @jennyhickson!
